app/controllers/ems_common.rb

```diff
@@ -264,6 +85,33 @@ def show
     end
   end
 
+  def calculate_display_class(display_name, session_display)
+    classes = [Container, ContainerReplicator, ContainerNode, ContainerGroup,
+               ContainerService, ContainerImage, ContainerRoute,
+               ContainerProject, ContainerImageRegistry, AvailabilityZone,
+               CloudTenant, CloudVolume, Flavor, SecurityGroup,]
+    names = classes.collect { |cls| cls.name.underscore.pluralize }
+    if names.include?(display_name)
+      return classes[names.index(display_name)]
+    elsif names.include?(session_display)
+      return classes[names.index(session_display)]
+    end
+    nil
```

Comment:
```ruby
classes.each do |klass|
  name = klass.name.underscore.pluralize
  return klass if [display_name, session_display].include?(name)
end
nil
```

Can be translated to

```ruby
classes.detect do |klass|
  name = klass.name.underscore.pluralize
  [display_name, session_display].include?(name)
  # or `name == display_name || name == session_display` - this one is more performant
end
```
It's essentially the same, but there is zero `if` statements, zero explicit `return`s and no hanging `nil`s.